Misc CI changes

LebedevRI opened the pull request.

LebedevRI committed:
Github actions: extract per-os jobs into reusable/callable workflows

kmilos commented on .github/workflows/CI.yml:
```diff
@@ -329,62 +84,16 @@ jobs:
             msys2: { msystem: MINGW64, arch: x86_64 }
```
FYI, MINGW64 is being sort of deprecated, UCRT64 is now the recommended MSYS2 default: https://www.msys2.org/news/#2022-10-29-changing-the-default-environment-from-mingw64-to-ucrt64

LebedevRI replied:
Good to know, but i'm not sure what to do about it, just dropping it in favor of the latter seems lossy?

LebedevRI committed:
Github actions: mess with windows msys2 jobs

LebedevRI committed:
Github actions: don't set name of jobs in called workflows

LebedevRI committed:
Github actions: split single monolithic CI job into groups

Note that there are no dependencies, that results in
large increase in total wall time.

LebedevRI committed:
Github actions: try to make job names better

LebedevRI committed:
Github actions: macOS: add xcode 14.1, bump 13.3.1 to 13.4.1 and move into legacy

LebedevRI committed:
Github actions: linux: split 'apt get update/install' into two steps, use eatmydata

LebedevRI committed:
CI: split build step into multiple smaller steps

LebedevRI committed:
CI: try to enable benchmarks build

LebedevRI committed:
Build benchmarks by default, those who do not want them should opt-out

LebedevRI committed:
CI: disable benchmarks build in clang SA and win coverage builds
